src/main/kotlin/com/zepben/evolve/services/network/translator/NetworkCimToProto.kt

```diff
@@ -653,6 +655,17 @@ fun Analog.toPb(): PBAnalog = toPb(this, PBAnalog.newBuilder()).build()
 fun Control.toPb(): PBControl = toPb(this, PBControl.newBuilder()).build()
 fun Discrete.toPb(): PBDiscrete = toPb(this, PBDiscrete.newBuilder()).build()
 
+/************ IEC61970 Base Protection ************/
+
+fun toPb(cim: CurrentRelay, pb: PBCurrentRelay.Builder): PBCurrentRelay.Builder =
+    pb.apply {
+        pb.currentLimit1 = cim.currentLimit1 ?: UNKNOWN_DOUBLE
+        cim.inverseTimeFlag?.let { inverseTimeFlag = it } ?: clearInverseTimeFlag()
```

Comment:
This is the first time we have a `Boolean?` field for a CIM class. Question: Is this the correct way of specifying a null boolean in a protobuf message?

Reply:
Update: Managed to get it to work using `pb.hasField` when translating back.

Reply:
ah this actually hasn't been well considered, as protobuf v3 doesn't have any notion of null for booleans, so the protobuf field will simply default to false. I think the easiest way to solve this is to make the field a `oneof` in protobuf. If we make it a oneof you can differentiate between not set, true, and false, as there should be a helper method on the proto to check if a `oneof` is set.

I've done a PR for this here: https://github.com/zepben/evolve-grpc/pull/63

search for oneof here for an example: https://itnext.io/protobuf-and-null-support-1908a15311b6